pkg/broker/logic.go

```
@@ -88,6 +205,57 @@ func (b *BusinessLogic) Update(request *osb.UpdateInstanceRequest, c *broker.Req
 	return &response, nil
 }
 
-func (b *BusinessLogic) ValidateBrokerAPIVersion(version string) error {
+func (b *BrokerLogic) ValidateBrokerAPIVersion(version string) error {
+	return nil
+}
+
+func generateHabitatObject(planID string) (*habv1beta1.Habitat, error) {
+	n, i, err := matchService(planID)
+	if err != nil {
+		return nil, err
+	}
+	// Generate Habitat object based on service.
+	hab := NewHabitat(n, i, 1) // TODO: Decide how many instances we should be running?
+
+	return hab, nil
+}
+
+func (b *BrokerLogic) deleteResources(planID string) error {
+	n, _, err := matchService(planID)
+	if err != nil {
+		return err
+	}
+
+	err = b.DeleteHabitat(n)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func matchService(planID string) (string, string, error) {
+	name := ""
+	image := ""
+
+	switch planID {
+	case "002341cf-f895-49f4-ba04-bb70291b895c":
+		name = "redis"
+		image = "kinvolk/osb-redis:latest" // TODO: find a better way then latest!
+	case "86064792-7ea2-467b-af93-ac9694d96d5b":
+		name = "nginx"
+		image = "kinvolk/osb-nginx:latest" // TODO: find a better way then latest!
+	case "":
```

Should we also have a `default` case, where we handle a GUID unknown to us?

---

I guess we could return an error for default when the plan ID is not known. 👍